core/finality-grandpa/src/environment.rs

```diff
@@ -500,15 +500,25 @@ where
 		let prevote_timer = Delay::new(now + self.config.gossip_duration * 2);
 		let precommit_timer = Delay::new(now + self.config.gossip_duration * 4);
 
-		let local_key = self.config.local_key.as_ref()
-			.filter(|pair| self.voters.contains_key(&pair.public().into()));
+		let local_key = crate::is_voter(&self.voters, &self.config.keystore);
```

Comment:
should return an iterator, so that we can vote with multiple keys if needed. @folsen if you're at a loose end, maybe you can do this (in a new branch :-))?